app/Jobs/ProcessMediaWikiJobsJob.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Log;
+use Maclof\Kubernetes\Client;
+use Maclof\Kubernetes\Models\Job as KubernetesJob;
+
+class ProcessMediaWikiJobsJob implements ShouldQueue, ShouldBeUnique
+{
+    use InteractsWithQueue, Queueable;
+
+    private string $wikiDomain;
+
+    public function __construct ( string $wikiDomain )
+    {
+        $this->wikiDomain = $wikiDomain;
+    }
+
+    public function uniqueId(): string
+    {
+        return $this->wikiDomain;
+    }
+
+    public function handle (Client $kubernetesClient): void
+    {
+        $mwPod = $kubernetesClient->pods()->setFieldSelector([
+            'status.phase' => 'Running'
+        ])->setLabelSelector([
+            'app.kubernetes.io/name' => 'mediawiki',
+            'app.kubernetes.io/component' => 'app-backend'
+        ])->first();
+
+        if (!$mwPod) {
+            $this->fail(
+                new \RuntimeException(
+                    'Unable to find a running MediaWiki pod in the cluster, '.
+                    'cannot continue.'
+                )
+            );
+            return;
+        }
+
+        $mwPod = $mwPod->toArray();
+
+        $jobSpec = new KubernetesJob([
+            'metadata' => [
+                'generateName' => 'run-all-mw-jobs-'
+            ],
+            'spec' => [
+                'ttlSecondsAfterFinished' => 60 * 60,
+                'template' => [
+                    'metadata' => [
+                        'name' => 'run-all-mw-jobs'
+                    ],
+                    'spec' => [
+                        'containers' => [
+                            0 => [
+                                'name' => 'run-all-mw-jobs',
+                                'image' => $mwPod['spec']['containers'][0]['image'],
+                                'env' => array_merge(
+                                    $mwPod['spec']['containers'][0]['env'],
+                                    [['name' => 'WBS_DOMAIN', 'value' => $this->wikiDomain]]
+                                ),
+                                'command' => [
```

Comment:
We could also externalise this script if we're unhappy with having it defined inline? Not sure.

Reply:
I wonder if we want to; right now this is "fine" but not the easiest to read. I'd probably prefer if we could do something like helmfile does and have templates for these yaml files ? No idea if this is doable though?

Reply:
Let's maybe keep it like this for now until we figure out the more important stuff.